zip-bolt-support.rst

```diff
@@ -0,0 +1,334 @@
+
+The customer picks an inital wallet public key ``<wpk>``.
+
+The customer and merchant escrow the necessary funds in a funding transaction, denoted ``escrow-tx``. 
+
+1.3 Closing a Channel: Overview
+-------------
+
+A customer should be able to close the channel by posting a *closing token* ``close-token``, which is a blind signature from the merchant under ``<MERCH-PK>`` on a special closing wallet that contains ``<cust-pk>, <wpk>, <balance-cust>, <balance-merch>, CLOSE``. We use ``cust-close-tx`` to denote the transaction posted by the customer to initiate channel closure.
+
+A merchant should be able to close the channel by either posting a special closing transaction ``merch-close-tx`` (detailed in Section 2.3.2) or, if the customer posts an outdated version of their closing token, a signed revocation token, ``revocation-token`` as detailed below. The revocation token ``revocation-token`` is a signature under the wallet public key ``<wpk>`` on the special revocation message ``<wpk> || REVOKED``. The transaction posted by the merchant to dispute is denoted ``dispute-tx``.
+
+The customer and merchant may also negotiate off-chain to form a *mutual close transaction*, ``mutual-close-tx``. Off-chain collaboration to create ``mutual-close-tx`` reduces the required number of on-chain transactions and eliminates the time delays.
+
+2. Transparent/Shielded Tx: Using T/Z-addresses and WTPs
+-------------
+
+We assume the following specific features are present:
+
+(1) Support for whitelisted transparent programs (WTPs) that enables 2-of-2 multi-sig style transactions
+(2) Can specify absolute lock time in transaction
+(3) Can specify relative lock time in transparent program
+(4) Can specify shielded inputs and outputs
+(5) A non-SegWit approach that fixes transaction malleability
+(6) ``OP_BOLT`` logic expressed as WTPs. We will use the Bolt WTPs defined in Section 2.1: ``open-channel``, ``cust-close``, and ``merch-close``.
+
+**Privacy Limitations**. The aggregate balance of the channel will be revealed in the funding transaction ``escrow-tx``. Similarly, the final splitting of funds will be revealed to the network. However, for channel opening and closing, the identity of the participants remains hidden. Channel opening and closing will also be distinguishable on the network due to use of WTPs.
+
+**Channel Opening**. The funding transaction ``escrow-tx`` spends ZEC from one or more shielded addresses to a transparent output that is encumbered by a Bolt transparent program. See Section 2.1 for what the funding transaction looks like when instantiated using WTPs.
+
+2.1 Bolt WTPs
+--------------
+
+Transparent programs take as input a ``predicate``, ``witness``, and ``context`` and then output a ``True`` or ``False`` on the stack. Bolt-specific transparent programs are deterministic and any malleation of the ``witness`` will result in a ``False`` output. The WTPs are as follows:
```

Comment:
```suggestion
Transparent programs take as input a ``predicate``, ``witness``, and ``context`` and then output a ``True`` or ``False`` on the stack. Bolt-specific transparent programs are deterministic and any malleation of the ``witness`` will result in a ``False`` output. The TZEs are as follows:
```
Make sure terminology is consistent with ZIP 222 throughout this section (e.g. the ZIP does not use "program"), and reference it.